### PR TITLE
Added Heroku support

### LINKED ISSUE #1: Need help
Hello, I am very fascinated with your project because so far only you have provided the code for this topic. I have a problem because I would like to implement this program on my wordpress website, but I do not know how to do it, could you help me please? I need this for me and my girlfriend I will be very grateful (I am willing to pay for your help). Take care and please tell me back asap.

### PR DESCRIPTION
**Pull request by anjannair**
This should fix #1 
The Heroku deploy button will only work after being merged because it requires the `app.json` in the repository. Here is [link](https://devcenter.heroku.com/articles/heroku-button) to the docs for that.

You can test the button here (this is linked to my repository as it contains the `app.json`)
[Deploy](https://heroku.com/deploy?template=https://github.com/anjannair/mood-meter)

Removed webhook url from config and added it to `.env` which user has to input. This will help with easy deployment to Heroku.

Edit - 
Removed `package-lock.json`. There is no use of keeping that in the repository as each user generates one when they install dependencies.

### CODE REVIEW

**Comment by filippofinke** [comment]
Thanks for the contribution!

But in order to be accepted I think we should transform the config.js file into environment variables because at the moment on Heroku it shows the error of renaming the configuration file "config.sample.js" to "config.js".

Wdyt?

**Comment by anjannair** [comment]
Oh yeah definitely! I do not think its necessary to keep the config as sample anymore too 

Give me a little more time I will revamp the `config.json` too

**Comment by anjannair** [comment]
@filippofinke can you check if this is working appropriately?
Do not merge yet as I am yet to integrate config.json completely into env

**Comment by anjannair** [comment]
@filippofinke alright feel free to poke around the code. I am unsure if you would like to include colors in the environment variables or not hence I have left that as it is.

If you are confused what is EJS just check out - https://ejs.co/ (PS - It is plain and simple HTML with a small twist and a different extension)

**Comment by anjannair** [comment]
@filippofinke final push done! You can now merge it!

P.S - The test button to deploy can be found in the main comment of this PR

**Comment by filippofinke** [comment]
I think that since you added some kind of server side rendering the fetch request for the mood is not needed anymore.

Also I think that also the emojis should be an env variable.

**Comment by anjannair** [comment]
> I think that since you added some kind of server side rendering the fetch request for the mood is not needed anymore.
> 
> Also I think that also the emojis should be an env variable.

So do you like the server side rendering or do you want me to remove it?

Should I add the emojis to the env variable?

File: config.json [hunk 1]
```diff
@@ -1,12 +1,6 @@
 {
   "PORT": 8080,
   "CONFIG": {
-    "colors": {
-      "--bg-color": "#fdf6f0",
-      "--text-color": "#a19882",
-      "--slider-thumb-color": "#cdbba7",
-      "--slider-bg-color": "linear-gradient(90deg, rgba(255, 0, 0, 1) 0%, rgba(227, 255, 0, 1) 50%, rgba(0, 255, 0, 1) 100%);"
-    },
     "emojis": [
```
Comment (filippofinke) [inline config.json:4]: We can remove the emojis array and create a new env variable.

File: public/js/index.js [hunk 2]
```diff
@@ -25,13 +25,6 @@ slider.addEventListener("change", ({ target: { value } }) => {
 window.addEventListener("load", async () => {
   config = await fetch("/config").then((r) => r.json());
```
Comment (filippofinke) [inline public/js/index.js:26]: We can remove this fetch for the cofing and instead use template variables from ejs.

File: public/js/index.js [hunk 3]
```diff
@@ -25,13 +25,6 @@ slider.addEventListener("change", ({ target: { value } }) => {
...
 
-  if (config.colors) {
-    let root = document.documentElement;
-    for (const [variable, value] of Object.entries(config.colors)) {
-      root.style.setProperty(variable, value);
-    }
-  }
-
   let mood = await fetch("/mood").then((r) => r.json());
```
Comment (filippofinke) [inline public/js/index.js:28]: Same here, we can remove the fetch request and use instead an ejs variable.

File: server.js [hunk 4]
```diff
@@ -30,7 +30,7 @@ app.use(express.static("public"));
 app.set("view engine", "ejs");
 
 app.get("/", (req, res) => {
-  res.render("index",{title:process.env.TITLE});
+  res.render("index",{title:process.env.TITLE,bgColor:process.env.bgColor,textColor:process.env.textColor,sliderThumbColor:process.env.sliderThumbColor,sliderBgColor:process.env.sliderBgColor});
```
Comment (filippofinke) [inline server.js:33]: Please, format the code correctly.

File: views/index.ejs [hunk 5]
```diff
@@ -21,12 +21,21 @@
...
     <div id="emojis" class="emojis"></div>
     <input min="0" step="0.1" id="slider" class="slider" type="range" />
   </div>
+  <script>
+    let root = document.documentElement;
+    root.style.setProperty("--bg-color", "<%=bgColor%>");
+    root.style.setProperty("--text-color", "<%=textColor%>");
+    root.style.setProperty("--slider-thumb-color", "<%=sliderThumbColor%>");
+    root.style.setProperty("--slider-bg-color", "<%=sliderBgColor%>");
+  </script>
```
Comment (filippofinke) [inline views/index.ejs:38]: I am not sure if this is the best way to do it. I have never used eJs before, there isn't another way to change the css with eJs?

**Review by filippofinke — CHANGES_REQUESTED** [review]

**Comment by filippofinke** [comment]
> > I think that since you added some kind of server side rendering the fetch request for the mood is not needed anymore.
> > Also I think that also the emojis should be an env variable.
> 
> So do you like the server side rendering or do you want me to remove it?
> 
> Should I add the emojis to the env variable?

Yes, we can keep the server side rendering and remove all the client fetches.

Yes, this way we remove the config.js file and use only env variables.

**Pull request closed without merging**